File: Problem/Easy/Problem_8/Solution.cpp
// C++ program using to calculate length of the string using recursion.

#include<iostream>

// Recursive function
int findLength(char * str)
{
    if (*str == '\0')
        return 0;
    return 1 + findLength(str + 1);
}

// Main function
int main()
{
    char str[] = "Nhan Pham";
    std::cout << "The length of string " << str << " is: " << findLength(str);
    return 0;
}

File: Problem/Easy/Problem_8/Solution.py
# Python3 program using to calculate length of the string using recursion.

# Recursive function
def findLength(s : str, length = 0) -> int:
    if s == '':
        return length
    return findLength(s[1:], length + 1)

# Main
if __name__ == '__main__':
    s = "Nhan Pham"
    print('Length of "{}" is: {}'.format(s, findLength(s)))